### PR TITLE
kicad(-unstable): bump to latest version, cleanup and fixes

### PR DESCRIPTION
**Pull request by evils**
kicad: 7.0.0 -> 7.0.1
kicad-unstable: 2023-02-14 -> 2023-03-29

with a bunch of cleanup and fixes
(most notably, a kicad-unstable build and ngspice tests fixes)
the removal of the pointless package option `withOCC`
and the addition of `requests` as a runtime dependency of some vendored plugins

###### Description of changes

https://www.kicad.org/blog/2023/03/KiCad-7.0.1-Release/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by evils** [comment]
changed to include a now up-to-date and fixed kicad-unstable

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 1]
```diff
@@ -97,48 +81,48 @@ stdenv.mkDerivation rec {
...
+
   cmakeFlags = [
-    # RPATH of binary /nix/store/.../bin/... contains a forbidden reference to /build/
-    "-DCMAKE_SKIP_BUILD_RPATH=ON"
     "-DKICAD_USE_EGL=ON"
-    "-DCMAKE_CTEST_ARGUMENTS='--exclude-regex;qa_eeschema'"  # upstream issue 12491
+    "-DOCC_INCLUDE_DIR=${opencascade-occt}/include/opencascade"
+  ]
+  ++ optionals (stable) [
+    # upstream issue 12491, should be resolved in the next release
```
Comment (RaitoBezarius) [inline pkgs/applications/science/electronics/kicad/base.nix:93]: can you put a link to it?

Comment (evils) [reply pkgs/applications/science/electronics/kicad/base.nix]: done

File: pkgs/applications/science/electronics/kicad/base.nix [hunk 2]
```diff
@@ -97,48 +81,49 @@ stdenv.mkDerivation rec {
...
+  ]
+  ++ optionals (stable) [
+    # https://gitlab.com/kicad/code/kicad/-/issues/12491
+    # should be resolved in the next release
+    "-DCMAKE_CTEST_ARGUMENTS='--exclude-regex;qa_eeschema'"
   ]
-  ++ optionals (withScripting) [
-    "-DKICAD_SCRIPTING_WXPYTHON=ON"
+  ++ optionals (!stable) [ # workaround for https://gitlab.com/kicad/code/kicad/-/issues/14346
+    "-DPYTHON_SITE_PACKAGE_PATH=${placeholder "out"}/lib/python${lib.versions.majorMinor python.version}/site-packages"
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/kicad/base.nix:98]:
```suggestion
    "-DPYTHON_SITE_PACKAGE_PATH=${placeholder "out"}/${python.sitePackages}/"
```

Comment (evils) [reply pkgs/applications/science/electronics/kicad/base.nix:98]: oh wow, here i was being happy there was a `majorMinor` function xD
thanks!

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**